### PR TITLE
chore: tiny refactor

### PR DESCRIPTION
**Pull request by u2**

### CODE REVIEW

File: core/src/block.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ pub(crate) fn cal_proposals_hash(vec: &[ProposalShortId]) -> H256 {
         let mut ret = [0u8; 32];
         let mut blake2b = new_blake2b();
         for id in vec.iter() {
-            blake2b.update(&(&id as &[u8; 10])[..]);
```
Comment (u2) [inline core/src/block.rs]: Any of `id as &[u8; 10]` and `id.deref()` is ok.

File: core/src/block.rs [hunk 2]
```diff
@@ -8,12 +8,13 @@ use hash::new_blake2b;
...
 use std::borrow::ToOwned;
+use std::iter::FromIterator;
+use std::ops::Deref;
 
 fn cal_transactions_root(vec: &[Transaction]) -> H256 {
     merkle_root(
         &vec.iter()
-            .map(Transaction::hash)
-            .map(ToOwned::to_owned)
+            .map(|x| Transaction::hash(x).to_owned())
```
Comment (quake) [inline core/src/block.rs]:
```suggestion
            .map(|tx| tx.hash().to_owned())
```

File: core/src/block.rs [hunk 3]
```diff
@@ -24,8 +25,7 @@ fn cal_witnesses_root(vec: &[Transaction]) -> H256 {
     witnesses.extend(
         vec.iter()
             .skip(1)
-            .map(Transaction::witness_hash)
-            .map(ToOwned::to_owned),
+            .map(|x| Transaction::witness_hash(x).to_owned()),
```
Comment (quake) [inline core/src/block.rs]:
```suggestion
            .map(|tx| tx.witness_hash().to_owned()),
```

**Comment by quake** [comment]
@u2 cargo fmt failed 🤓

**Comment by doitian** [comment]
You can setup the pre-commit hook locally.

https://github.com/nervosnetwork/ckb/blob/develop/devtools/git/pre-commit

**Comment by u2** [comment]
thanks @doitian , @quake

**Comment by doitian** [comment]
Linters failed because of a cache issue, a possible fixing is submit in https://github.com/nervosnetwork/ckb/pull/1004

**Pull request merged**